plugins/modules/rds_cluster.py

```diff
@@ -1102,6 +1129,29 @@ def ensure_present(cluster, parameters, method_name, method_options_name):
     return changed
 
 
+def handle_remove_from_global_db(module, cluster):
+    global_cluster_id = module.params.get("global_cluster_identifier")
+    db_cluster_id = module.params.get("db_cluster_identifier")
+    db_cluster_arn = cluster["DBClusterArn"]
+
+    if module.check_mode:
+        return True
+
+    try:
+        client.remove_from_global_cluster(DbClusterIdentifier=db_cluster_arn, GlobalClusterIdentifier=global_cluster_id)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(
+            e, msg=f"Failed to remove cluster {db_cluster_id} from global DB cluster {global_cluster_id}."
+        )
+
+    # this is needed as even though we have wait_for_cluster_status, it takes a few seconds
+    # for cluster to change status from 'available' to 'promoting'
+    time.sleep(15)
+    wait_for_cluster_status(client, module, db_cluster_id, "cluster_available")
```

Comment:
With this change, the user has to wait no matter what even if there is no change to apply to the cluster or maybe they do not want to wait. Maybe I am wrong, but it would perhaps be more efficient, if there is no change to apply to the cluster, but only to remove from the global db and the user does not want to wait, just remove it without any waiting. On the other hand, if there is something to be updated or if there is nothing to be updated and the user wants to wait for the cluster to be available, in that case you have to wait. I might be wrong, but since rds_cluster is s slow module, I guess we should probably minimise the waiting times when possible, this is my thought.

Reply:
imo, 

1. the part `time.sleep(15)`, is much needed as without the sleep time, integration tests return false failure due to time it takes for clusterto go `from available to promoting` state.

2. the `wait_for_cluster_status(client, module, db_cluster_id, "cluster_available")` is also something I would like to have, as if there is any changes to apply to cluster along with removing it from global db, the changes won't get applied while the cluster is in `promoting` state.

I could be wrong though. cc @hakbailey @jillr

Reply:
What I'm suggesting is to wait only when needed.

Reply:
okay, I'll change it to wait when `wait` param is set `true`
https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_cluster.py#L60

Reply:
@jillr @tremble  any thoughts?

Reply:
We chatted about this and I think we can add a custom waiter to https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/waiters.py that would cover this scenario without having a have a static sleep.  @mandar242 will give that a try,

Reply:
@jillr @alinabuzachis
Added a custom waiter that - 'waits for promoting initiates on AWS end, i.e. cluster status changes from `available` to `promoting`'.
https://github.com/ansible-collections/amazon.aws/pull/1705/commits/eedaa256c9c39bfb745f4f06b95261772e6c12d2